task.ps1

```diff
@@ -81,4 +81,21 @@ New-AzVm `
 -PublicIpAddressName $jumpboxVmName
 
 
-# Write your code here  -> 
+# # Write your code here  ->
+
+# Add Private DNS Zone
+Write-Host "Creating a private DNS zone ..."
+$privateDnsZone = New-AzPrivateDnsZone -ResourceGroupName $resourceGroupName -Name $privateDnsZoneName
+
+Write-Host "Linking private DNS zone to virtual network ..."
+New-AzPrivateDnsVirtualNetworkLink -ResourceGroupName $resourceGroupName -ZoneName $privateDnsZoneName `
+-VirtualNetworkId $virtualNetwork.Id -Name "$($virtualNetworkName)-dnslink" -EnableRegistration
+
+# Add CNAME Record
+Write-Host "Creating CNAME record in the private DNS zone ..."
+$cnameRecord = New-AzPrivateDnsRecordSet -Name "todo" -RecordType CNAME -ZoneName $privateDnsZoneName -ResourceGroupName $resourceGroupName
+Add-AzPrivateDnsRecordConfig -RecordSet $cnameRecord -Cname "$webVmName.$privateDnsZoneName"
+Set-AzPrivateDnsRecordSet -RecordSet $cnameRecord
+
+Write-Host "Public IP Address for Jumpbox: $($publicIP.IpAddress)"
```

Comment:
Recommend to review section where you create Public IP. You missed important part for getting value from here :)

Reply:
Public IP created by the task author.
I just deleted the 100th line

Reply:
I want to pay your attention that you delete variable. I understand that Public IP was created. However, you delete variable you could not use it in 100 line